src/Type.cpp

```diff
@@ -120,6 +127,13 @@ bool Type::can_represent(int64_t x) const {
         return x >= min_int(bits()) && x <= max_int(bits());
     } else if (is_uint()) {
         return x >= 0 && (uint64_t)x <= max_uint(bits());
+    } else if (is_bfloat()) {
+        switch (bits()) {
+        case 16:
+            return (int64_t)(float)(bfloat16_t)(float)x == x;
```

Comment:
I am speechless

Reply:
Haha yeah - I was copying the original code. The idea is to round-trip from int64 to bfloat16 then back to int64 to see if the value was preserved. The bfloat16 <-> int64 conversion must be done via float in both directions though.

Reply:
`(int64_t)(float)(bfloat16_t)(float)` is my next band name